tests/test_subparser.py

```diff
@@ -0,0 +1,263 @@
+from dataclasses import dataclass, field
+from typing import Union
+from sys import version_info
+
+from pytest import mark, raises
+
+from pydargs import parse
+
+
+@dataclass
+class Action1:
+    a: int
+    b: str = "abc"
+
+
+@dataclass
+class Action2:
+    c: int = 42
+    b: str = "abc"
+    e: str = field(default="def", metadata=dict(positional=True))
+
+
+@dataclass
+class Action3:
+    d: float
+    e: list[str] = field(default_factory=lambda: ["def"])
+
+
+@dataclass
+class Action4:
+    sub_action: Union[Action1, Action2]
+    string4: str = "four"
+
+
+class TestParseAction:
+    @dataclass
+    class Config:
+        action: Union[Action1, Action2]
+        var: int = 12
+        flag: bool = field(default=False, metadata=dict(as_flags=True))
+
+    @mark.parametrize(
+        "help_string",
+        [
+            "usage: prog [-h] [--var VAR] [--flag | --no-flag]",
+            "{Action1,action1,Action2,action2} ...",
+            "action:  {Action1,action1,Action2,action2}",
+        ],
+    )
+    def test_help(self, capsys, help_string: str):
```

Comment:
why is this done with parametrize instead of defining this inside the test?

Reply:
I don't think it matters much. Would you prefer a loop inside the test?

Reply:
Ah no I thought it was one big string you wanted to compare but had to split it up for some reason